pynapple/process/signal_processing.py

```diff
@@ -0,0 +1,234 @@
+"""
+Signal processing tools for Pynapple.
+
+"""
+
+from itertools import repeat
+
+import numpy as np
+
+import pynapple as nap
+
+
+def compute_fft(sig, fs):
+    """
+
+    Parameters
+    ----------
+    sig : pynapple.Tsd
+        The signal time series to obtain time frequency decomposition from
+    fs : int / float
```

Comment:
I would write `fs=None`. If None, compute `fs` by averaging the diff between two time indices. We need to write somewhere that this functions assume constant sampling rate.